packages/framework/src/types/schema.types.ts

```diff
@@ -3,14 +3,17 @@ import zod from 'zod';
 
 export type JsonSchema = JSONSchema;
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type ClassType<T = any> = new (...args: any[]) => T;
```

Comment:
Could this be converted to `unknown[]` rather than `any[]`?

Reply:
my ide was not able to resolve the typings in this case :(
also several other places have no working typings, but i guess this is not a problem introduced with class-validator.
Take look at the test suites, here you see what Else is failing.

Reply:
It seems the `FromSchema` inference was causing the troubles. I've separated into a `ClassType` and `InferenceClassProperties` generic and we are golden! Fix coming here soon.